contrib/opencensus-ext-azure/opencensus/ext/azure/metrics_exporter/__init__.py

```diff
@@ -35,49 +35,66 @@
 logger = logging.getLogger(__name__)
 
 
-class MetricsExporter(object):
+class MetricsExporter(TransportMixin):
```

Comment:
`MetricsExporter` isn't a `BaseExporter` because we're using the export thread from `get_exporter_thread` instead of `BaseExporter`'s worker?

It looks like there's a fair amount of duplicated code here, and that `BaseExporter` should be refactored or made specific to tracing.

Reply:
Yes it seems like `MetricsExporter` and `AzureLogHandler` have very similar behaviour in terms of the exporter thread. I would like to eventually have all of them be a `BaseExporter`. Was there any discussions with you and Reiley why this shouldn't be the case (why it was like this in the first place)? In any case, I'd like to keep those refactoring changes separate from a feature PR like this one, but I will cut a ticket open for it.

Reply:
[#852]

Reply:
> Was there any discussions with you and Reiley why this shouldn't be the case (why it was like this in the first place)?

It looks like `BaseExporter` was factored out in #642, but that PR didn't touch the `StackdriverStatsExporter`, which influenced the design of the other stats exporters. In any case this can be a problem for another PR.